File: backend/src/main/java/com/inventory/dto/ProductDTO.java
package com.inventory.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductDTO {
    private Long id;
    
    @NotBlank(message = "Product name is required")
    private String name;

    private String code;
    
    @NotNull(message = "Product value is required")
    @Positive(message = "Product value must be positive")
    private BigDecimal value;
    
    private List<ProductMateriaPrimaDTO> materiasPrimas;
}

File: backend/src/main/kotlin/com/inventory/dto/ProductDTO.kt
package com.inventory.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import java.math.BigDecimal

data class ProductDTO(
    var id: Long? = null,
    
    @field:NotBlank(message = "Product name is required")
    var name: String = "",
    
    @field:NotNull(message = "Product value is required")
    @field:Positive(message = "Product value must be positive")
    var value: BigDecimal = BigDecimal.ZERO,
    
    var rawMaterials: List<ProductRawMaterialDTO>? = null
)
